### PR TITLE
Seen phrase

### PR DESCRIPTION
**Pull request by mitia-anah**
Hello team!

The seen phrases is done, so you can have a look at it. Happy to see your comment.
Thanks

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,6 +28,7 @@
     "react-native-vector-icons": "^8.1.0",
     "react-redux": "^7.2.4",
     "redux": "^4.1.0",
+    "redux-persist": "^6.0.0",
```
Comment (andrzej-onja) [inline package.json]: Do you need that ?

File: src/screens/Learning.js [hunk 2]
```diff
@@ -63,20 +67,25 @@ export default ({
     item => {
       if (item.id === currentPhrase.id) {
         // TODO add to learned
+        const seenPhrasesRemoved = seenPhrases.filter(
+          phrase => phrase.id !== item.id,
+        );
```
Comment (andrzej-onja) [inline src/screens/Learning.js]: move this logic inside `removeFromSeenPhrases` action and as a payload pass just `item`

File: src/redux/actions/index.js [hunk 3]
```diff
@@ -34,3 +39,65 @@ export function setLanguageName(language) {
...
+    } else {
+      itemsToStore = [...storedSeenPhrases, phrase];
+    }
+    await setDataToStorage(SEEN_PHRASES_KEY, itemsToStore);
+    dispatch(setSeenPhrases(itemsToStore));
+    return Promise.resolve();
+  };
+}
+
+export function removeFromSeenPhrases(seenPhrases) {
```
Comment (andrzej-onja) [inline src/redux/actions/index.js]: rename seenPhrase_s_ to seenPhrase

File: src/redux/reducers/index.js [hunk 4]
```diff
@@ -46,10 +51,35 @@ function nativeLanguage(state = '', action) {
...
+function seenPhrases(state = [], action) {
+  switch (action.type) {
+    case SET_SEEN_PHRASE:
+      return action.payload;
+    case SEEN_PHRASES_KEY:
+      return action.payload;
+    case REMOVE_FROM_SEEN_PHRASES:
+      return action.payload;
+    case SET_DISABLED_SEEN_PHRASE_AREA:
+      return action.payload;
```
Comment (andrzej-onja) [inline src/redux/reducers/index.js]: all of the above can be replaced with SET_SEEN_PHRASE as it does the same

File: src/screens/Home.js [hunk 5]
```diff
@@ -130,41 +145,40 @@ export default ({
...
           <View style={styles.heading}>
-            <SectionHeading text={seenPhrasesHeading} />
+            <SectionHeading text="Seen phrases:" />
           </View>
           <List
-            data={[{id: 1, name: '35 words and phrases'}]}
-            text={learnButtonText}
             lang={nativeLanguage}
-            data={seenPhrases}
+            data={[{ id: 1, name: seenPhrasesTotal() }]}
```
Comment (Fodilahy-mena) [inline src/screens/Home.js]: instead of id: 2 use something like id: ###seen-phrases### just for consistency.

Comment (Fodilahy-mena) [reply src/screens/Home.js]: the commit message `adds csome chsnges` should be changed. [See here on how to change commit messages](https://linuxize.com/post/change-git-commit-message/)

Comment (Fodilahy-mena) [reply src/screens/Home.js]: the commit message `adds csome chsnges` should be changed. [See here on how to change commit messages](https://linuxize.com/post/change-git-commit-message/)

File: src/containers/LearningScreenContainer.js [hunk 6]
```diff
@@ -1,21 +1,32 @@
...
+  isSeenPhrases
 } from '../redux/selectors';
 
 function mapStateToProps(state) {
   return {
+    categories: categoriesRoot(state),
     categoryPhrases: categoryPhrasesRoot(state),
     currentCategoryName: currentCategoryName(state),
     nativeLanguage: nativeLanguageRoot(state),
+    isSeenPhrases: isSeenPhrases(state),
```
Comment (andrzej-onja) [inline src/containers/LearningScreenContainer.js:20]: this is unecessary

File: src/redux/reducers/index.js [hunk 7]
```diff
@@ -46,6 +50,22 @@ function nativeLanguage(state = '', action) {
...
+function seenPhrases(state = [], action) {
+  switch (action.type) {
+    case SET_SEEN_PHRASE:
+      return action.payload;
+    // We need the seen phrases key type
+    case SEEN_PHRASES_KEY:
+      return action.payload;
+    // Removes phrases from the seen phrases  
+    case REMOVE_FROM_SEEN_PHRASES:
+      return action.payload;
```
Comment (andrzej-onja) [inline src/redux/reducers/index.js]: delete REMOVE_FROM_SEEN_PHRASES and SEEN_PHRASES_KEY they can be just replaced by SET_SEEN_PHRASE

File: src/redux/constants/index.js [hunk 8]
```diff
@@ -2,5 +2,8 @@
 export const SET_CATEGORIES = 'SET_CATEGORIES';
 export const SET_PHRASES = 'SET_PHRASES';
 export const SET_LANGUAGE_NAME = 'SET_LANGUAGE_NAME';
-export const SET_CURRENT_CATEGORY = 'SET_CURRENT_CATEGORY';
+export const SET_CURRENT_CATEGORY = 'SET_CURRENT_CATEGORY'; 
+export const SET_SEEN_PHRASE = 'SET_SEEN_PHRASE';
+export const SEEN_PHRASES_KEY = 'SEEN_PHRASES_KEY';
```
Comment (andrzej-onja) [inline src/redux/constants/index.js]: SEEN_PHRASES_KEY should be stored in the `src/utils/storage.js `

File: src/redux/selectors/index.js [hunk 9]
```diff
@@ -18,16 +19,27 @@ export const currentCategory = createSelector(
...
 );
+
+export const isSeenPhrases = createSelector(
+  [currentCategoryIdRoot],
+  (selectedCategoryId) => {
+    const selectedCategory = "###seenPhrases###" === selectedCategoryId
+    return selectedCategory;
+  },
+);
+
```
Comment (andrzej-onja) [inline src/redux/selectors/index.js:30]: Nice use of selector!

File: src/screens/Home.js [hunk 10]
```diff
@@ -50,12 +52,15 @@ export default ({
...
     setCategories(categories);
+    synchronizeStorageToRedux()
   }, []);
 
   const openCategoryPhrases = item => {
     const categoryId = item.id;
     setCurrentCategory(categoryId);
     // fetch Phrases for category
+    const phrasesForCategory = getPhrasesForCategoryId(item.id);
+    setPhrases(phrasesForCategory);
```
Comment (andrzej-onja) [inline src/screens/Home.js]: Conflict resolving mistake ?
`
const phrasesForCategory = getPhrasesForCategoryId(item.id);
    setPhrases(phrasesForCategory)
`
is duplicated

File: src/screens/Learning.js [hunk 11]
```diff
@@ -62,21 +66,23 @@ export default ({
...
       setDisableAllOptions(true);
-
       const answerOptionsWithSelected = answerOptions.map(phrase => {
-        return {...phrase, isSelected: phrase.id === item.id};
+        return { ...phrase, isSelected: phrase.id === item.id };
       });
 
       setAnswerOptions(answerOptionsWithSelected);
     },
     [currentPhrase, setDisableAllOptions, answerOptions],
```
Comment (andrzej-onja) [inline src/screens/Learning.js]: seenPhrases has to be added to dependency array
[currentPhrase, setDisableAllOptions, answerOptions, seenPhrases]

File: yarn.lock [hunk 12]
```diff
@@ -1422,10 +1422,17 @@
     prop-types "^15.6.1"
     react-lifecycles-compat "^3.0.4"
 
+<<<<<<< HEAD
+"@react-native-community/async-storage@^1.12.1":
+  version "1.12.1"
+  resolved "https://registry.yarnpkg.com/@react-native-community/async-storage/-/async-storage-1.12.1.tgz#25f821b4f6b13abe005ad67e47c6f1cee9f27b24"
+  integrity sha512-70WGaH3PKYASi4BThuEEKMkyAgE9k7VytBqmgPRx3MzJx9/MkspwqJGmn3QLCgHLIFUgF1pit2mWICbRJ3T3lg==
+=======
```
Comment (andrzej-onja) [inline yarn.lock]: There are some leftovers from rebasing 
<<<<<<< HEAD
=======

**Review by andrzej-onja — CHANGES_REQUESTED** [review]

File: src/containers/HomeScreenContainer.js [hunk 13]
```diff
@@ -1,30 +1,35 @@
...
     newTerms: newTermsRoot(state),
   };
 }
+
 const mapDispatchToProps = {
   setCategories,
   setCurrentCategory,
   setPhrases,
   setLanguageName,
+  setSeenPhrases,
```
Comment (constantingoeldel) [inline src/containers/HomeScreenContainer.js]: Why does the homescreen need this?

File: src/redux/actions/index.js [hunk 14]
```diff
@@ -63,11 +99,17 @@ export function addNewTerm(newTerm) {
...
+    }
+    
+    const storedSeenPhrases = await getDataFromStorage(SEEN_PHRASES_KEY);
+    if (!storedSeenPhrases) {
       return Promise.resolve();
     }
-    dispatch(setNewTerms(storedNewTerms));
+    dispatch(setSeenPhrases(storedSeenPhrases));
     return Promise.resolve();
   };
```
Comment (andrzej-onja) [inline src/redux/actions/index.js:112]: Lets rewrite it this way:
 
   ```
    const storedSeenPhrases = await getDataFromStorage(SEEN_PHRASES_KEY);
    if (storedSeenPhrases) {
      dispatch(setSeenPhrases(storedSeenPhrases));
    }
    
    return Promise.resolve();
```

File: src/redux/reducers/index.js (hunk 7 above)
Comment (andrzej-onja) [reply src/redux/reducers/index.js]: delete REMOVE_FROM_SEEN_PHRASES

**Review by andrzej-onja — APPROVED** [review]

**Review by Fodilahy-mena — APPROVED** [review]

**Review by rinonTen — APPROVED** [review]

**Pull request merged**